examples/movie_view_ratings.py

```diff
@@ -14,100 +14,133 @@
 
    or by other way to get a subset of lines from the dataset.
 
-4. Run python movie_view_ratings.py --input_file=<path to data.txt from 3> --output_file=<...>
+4. Run python movie_view_ratings.py --framework=<framework> --input_file=<path to data.txt from 3> --output_file=<...>
 """
 
 from absl import app
 from absl import flags
 import apache_beam as beam
 from apache_beam.runners.portability import fn_api_runner
+import pyspark
 from dataclasses import dataclass
 import pipeline_dp
 
 FLAGS = flags.FLAGS
 flags.DEFINE_string('input_file', '', 'The file with the movie view  data')
 flags.DEFINE_string('output_file', None, 'Output file')
+flags.DEFINE_enum('framework', None, ['beam', 'spark'], 'Pipeline framework to use.')
 
 
 @dataclass
 class MovieView:
-  user_id: int
-  movie_id: int
-  rating: int
+    user_id: int
+    movie_id: int
+    rating: int
 
 
-def parse_line(line, movie_id):
-  # 'line' has format "user_id,rating,date"
-  split_parts = line.split(',')
-  user_id = int(split_parts[0])
-  rating = int(split_parts[1])
-  return MovieView(user_id, movie_id, rating)
-
+def calc_dp_rating_metrics(movie_views, ops):
+    """Computes dp metrics."""
 
-class ParseFile(beam.DoFn):
+    # Set the total privacy budget.
+    budget_accountant = pipeline_dp.BudgetAccountant(epsilon=1, delta=1e-6)
 
-  def __init__(self):
-    self.movie_id = -1
+    # Specify a pipeline framework to use.
+    ops = pipeline_dp.BeamOperations()
```

Comment:
this line overrides "ops" from arguments, please remove it

Reply:
Good catch! Not sure how I managed to get the Spark run produce the output while testing the code.